infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -787,9 +784,11 @@ private LogData  readRecord(SegmentHandle sh, long address)
             } else {
 
                 try {
-                    FileChannel channel = FileChannel.open(FileSystems.getDefault().getPath(filePath),
-                            EnumSet.of(StandardOpenOption.READ, StandardOpenOption.WRITE,
-                                    StandardOpenOption.CREATE_NEW, StandardOpenOption.SPARSE));
+                    EnumSet<StandardOpenOption> options = EnumSet.of(
+                            StandardOpenOption.READ, StandardOpenOption.WRITE,
+                            StandardOpenOption.CREATE_NEW, StandardOpenOption.SPARSE
+                    );
+                    FileChannel channel = FileChannel.open(FileSystems.getDefault().getPath(filePath), options);
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use try-with-resources or close this "FileChannel" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2095)